starter-code/js/index.js

```diff
@@ -1,48 +1,129 @@
 function deleteItem(e){
-
+  var productToDelete = e.currentTarget.parentNode.parentNode;
+  var parentOfProduct = productToDelete.parentNode;
+  parentOfProduct.removeChild(productToDelete);
 }
 
 function getPriceByProduct(itemNode){
+  /* console.log(itemNode);
+    var productPrice = Number(itemNode.innerText);
+    console.log(productPrice);
+    var quantity = document.querySelectorAll('.quantity')[i].valueAsNumber;
+    var totalField = document.querySelectorAll('.total-value')[i];
+    var total = productPrice * quantity; 
+
 
+  return total; */
 }
 
 function updatePriceByProduct(productPrice, index){
 
 }
 
 function getTotalPrice() {
-
+  var numOfProducts = document.querySelectorAll('.product-list-item');
+  var totalCartValue = 0;
+  for (var i = 0; i < numOfProducts.length; i++) {
+    var productPrice = Number(document.querySelectorAll('.value')[i].innerText);
```

Comment:
Instead of doing a very big-heavy function, break it in little functions that only do one thing.

Reply:
Instead of searching things in the document inside an iteration, you can create a variable with an array, and iterate over this array (it will make the iteration faster).